### PR TITLE
OP-18616: APIs to create/update/get/delete auth providers

### PR DESCRIPTION
**Pull request by rahul-chekuri**
https://devopsmx.atlassian.net/browse/OP-18469
https://devopsmx.atlassian.net/browse/OP-18616
https://devopsmx.atlassian.net/browse/OP-18369

UTC:
Added new ldap auth config
Modified ldap auth config
Disabled ldap auth
Deleted ldap auth
Above cases with saml too
Retrieved ldap and saml configs 

Unit test cases will be addressed as part of https://devopsmx.atlassian.net/browse/OP-19033

### CODE REVIEW

File: docker_build/gate.yml [hunk 1]
```diff
@@ -288,36 +295,6 @@ keycloak:
             defaultValue: http://localhost:8080/realms/isd-poc-realm/saml/endpoint
             loadValues: NA
 
-          - name: serviceProviderEntityId
```
Comment (rahul-chekuri) [inline docker_build/gate.yml:291]: these values will be imported from metadata

File: gate-web/gate-web.gradle [hunk 2]
```diff
@@ -47,13 +47,15 @@ dependencies {
   implementation "com.graphql-java-kickstart:graphql-spring-boot-starter:7.0.1"
   implementation "com.graphql-java-kickstart:graphql-java-tools:6.0.2"
 
-  implementation "io.springfox:springfox-swagger2"
+  implementation "io.springfox:springfox-swagger2:2.9.2"
```
Comment (rahul-chekuri) [inline gate-web/gate-web.gradle:50]: added swagger deps

File: gate-web/gate-web.gradle [hunk 3]
```diff
@@ -47,13 +47,15 @@ dependencies {
...
+  implementation "io.springfox:springfox-swagger-ui:2.10.0"
 
   implementation 'org.springframework.boot:spring-boot-starter-cache'
   implementation group: 'com.github.ben-manes.caffeine', name: 'caffeine', version: '2.9.2'
 
 
   implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign', version: '2.2.4.RELEASE'
+  implementation group: 'org.keycloak', name: 'keycloak-admin-client', version: '20.0.2'
```
Comment (rahul-chekuri) [inline gate-web/gate-web.gradle:58]: keycloak-admin-client provides an admin client. With this, we can access keycloak conveniently.

File: gate-web/src/main/java/com/netflix/spinnaker/gate/config/SwaggerConfig.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.netflix.spinnaker.gate.config;
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/netflix/spinnaker/gate/config/SwaggerConfig.java:1]: Enable Swagger

File: docker_build/gate.yml [hunk 5]
```diff
@@ -126,6 +126,13 @@ management:
       show-components: always
 
 keycloak:
+  realm: myrealm
+  ldapName: ldap
+  samlName: saml
```
Comment (rahul-chekuri) [inline docker_build/gate.yml:131]: Configured names so that we can easily integrate with existing keycloak.

**Review by Pranav-b-7 — APPROVED** [review]

**Pull request merged**